stdlib/public/core/MutableCollection.swift

```diff
@@ -372,6 +372,30 @@ extension MutableCollection {
   }
 }
 
+extension MutableCollection where Self: BidirectionalCollection {
+  public subscript(bounds: Range<Index>) -> MutableBidirectionalSlice<Self> {
+    get {
+      _failEarlyRangeCheck(bounds, bounds: startIndex..<endIndex)
+      return .init(base: self, bounds: bounds)
+    }
+    set {
+      _writeBackMutableSlice(&self, bounds: bounds, slice: newValue)
+    }
+  }
+}
+
+extension MutableCollection where Self: RandomAccessCollection {
+  public subscript(bounds: Range<Index>) -> MutableRandomAccessSlice<Self> {
+    get {
+      _failEarlyRangeCheck(bounds, bounds: startIndex..<endIndex)
+      return .init(base: self, bounds: bounds)
```

Comment:
I guess gyb makes the `.init` thing more appealing since you have to return an ugly gybbed typename like `RangeReplaceable${capability}Slice` (not in this example, but in some others).

Re the working of `.xxx`... that seems like just something to get over, since it's pervasive – this is just one more example. Sure, `CapitalizedName(...)` means creation but I think the bare `.init` does make it clearer what it's doing "at a glance" once you're used to it, to me at least, and also eliminates any chance of there being an additional subtype conversion you could be overlooking.